ethcore/src/client/client.rs

```diff
@@ -1457,23 +1596,28 @@ impl BlockChainClient for Client {
 
 	fn block_hash(&self, id: BlockId) -> Option<H256> {
 		let chain = self.chain.read();
-		Self::block_hash(&chain, &self.miner, id)
+		Self::block_hash(&chain, id)
 	}
 
-	fn code(&self, address: &Address, id: BlockId) -> Option<Option<Bytes>> {
-		self.state_at(id).and_then(|s| s.code(address).ok()).map(|c| c.map(|c| (&*c).clone()))
+	fn code(&self, address: &Address, state: StateOrBlock) -> Option<Option<Bytes>> {
```

Comment:
I think it was already mention, but we should replace `Option<Option<...>>` with something more descriptive. Currently it's quite confusing

Reply:
Yep, we have discussed it in https://github.com/paritytech/parity/issues/7098. Still I think, we need to refactor one thing at a time. Will probably return to it later, when dust settles.